src/image_service.cpp

```diff
@@ -415,12 +438,12 @@ ImageFile *ImageService::create_image_file(const char *config_path) {
         defaultDlCfg.HasMember("download")) {
         cfg.AddMember("download", defaultDlCfg["download"], cfg.GetAllocator());
     }
-    std::string accelerate_url = "";
+
     if (global_conf.p2pConfig().enable() &&
         check_accelerate_url(global_conf.p2pConfig().address())) {
-        accelerate_url = global_conf.p2pConfig().address();
+        std::string accelerate_url = global_conf.p2pConfig().address();
+        ((RegistryFS*)(global_fs.remote_fs))->setAccelerateAddress(accelerate_url.c_str());
     }
-    ((RegistryFS*)(global_fs.remote_fs))->setAccelerateAddress(accelerate_url.c_str());
```

Comment:
Fix a bug here. When using the cache, the type of remote_fs is CachedFs, and errors may occur if forced type conversion is performed.